docs/reference/mapping/types/runtime.asciidoc

```diff
@@ -0,0 +1,365 @@
+in the mapping and access it at search time without indexing your data during
+ingest time.
+
+Runtime fields beta[] are accessible from the search API like any other field that has
+<<doc-values,doc_values>> and is searchable. You can retrieve and query these
+fields, as well as aggregate on them.
+
+Runtime fields help to alleviate several common issues when using {es}:
+
+* Reindexing your data between development iterations is slow and can make
+experimenting on large datasets difficult
+* Indexing data before searching makes running one-off searches costly and
+resource intensive
+* Indexing all of your data instead of just the fields you want to search
+requires more disk space to gain search time performance
+* Reindexing data for time-based indices to ensure that existing indices
+include any new fields in the index template is slow
+* Determining how a field is evaluated in {kib} at index or search time is
+difficult for scripted fields because they have different needs based on where
+they are implemented
+
+Because runtime fields aren't indexed, you can more quickly ingest raw data
+into the Elastic Stack and immediately access it. By dynamically evaluating
+runtime fields at search time, you can optimize disk space by choosing which
+fields to index. You can also fix errors in indexed fields by overriding them
+with runtime fields, rather than reindexing all of your data. If you later
+decide that a runtime field is increasingly used for filtering or aggregation,
+you can remove the runtime field from the mapping and index the field to gain
+faster search speed.
+
+Runtime fields incur performance costs at search time, depending
+on the <<runtime-params-runtime-type,`runtime_type`>>. For example, let's say
+you created an anomaly detection job that operates on the `@timestamp` field.
+If the `@timestamp` field is a runtime field, the search cost would be extremely
+high because the data isn't indexed and {es} must compute the value for each
+document that matches the query.
+
+Runtime fields are useful when working with log data, especially when you're
+unsure about the data structure. Your search speed decreases, but your index
+size is much smaller and you can more quickly process logs without having to
+index them.
+
+[[runtime-mapping-fields]]
+==== Mapping a runtime field
+When mapping a runtime field, indexing is disabled by default. You map runtime
+fields by <<modules-scripting-using,defining a painless script>>. At search
+time, the script runs and generates values for each scripted field. Runtime
+scripts have access to the entire context of a document, the original
+`_source`, and the mapped field plus its values (`doc_values`).
+
+IMPORTANT: Updating a script while a query that relies on the script is running
+can return inconsistent results.
++
+Additionally, existing queries or visualizations that rely on runtime fields
+can break if scripts are updated.
```

Comment:
I chatted with Stacey and CJ about this, and they both indicated that users will be unaware in Kibana about changes to the underlying `runtime_type` -- at least for the first release. However, I think we can explain a bit more that changing the `runtime_type` will break a visualization:
>Existing queries or visualizations in Kibana that rely on runtime fields can
fail if you change the `runtime_type`. For example, a bar chart visualization
that uses a runtime field of `ip` would fail if the `runtime_type` is changed
to `boolean`.

Reply:
Makes sense to me, the example makes it clear what type of change and how things would fail, thanks!